src/main/java/org/folio/list/services/UserFriendlyQueryService.java

```diff
@@ -187,41 +190,30 @@ private Object getConditionValue(FieldCondition<?> fieldCondition, EntityType en
       .findFirst()
       .map(Field::getDataType)
       .orElseThrow();
+
     if (!(dataType instanceof DateType)) {
       return fieldCondition.value();
     }
-    LocalDateTime dateTime;
-    String timezone;
-    String dateString = (String) fieldCondition.value();
-    if (dateString.matches(DATE_REGEX)) {
-      dateTime = LocalDate.parse(dateString, DATE_FORMATTER).atStartOfDay();
-    } else if (dateString.matches(DATE_TIME_REGEX)) {
-      dateTime = LocalDateTime.parse(dateString, DATE_TIME_FORMATTER);
+
+    ZoneId tenantTimezone = configurationClient.getTenantTimezone();
+
+    return translationService.formatString(
+      tenantTimezone,
+      "{value, date, short}", "value",
+      handleDateValue((String) fieldCondition.value(), tenantTimezone)
+    );
+  }
+
+  private Instant handleDateValue(String value, ZoneId timezone) {
+    String dateString = value;
+    if (value.matches(DATE_REGEX)) {
+      return LocalDate.parse(value).atStartOfDay(timezone).toInstant();
+    } else if (value.matches(DATE_TIME_REGEX)) {
+      return Instant.from(DATE_TIME_FORMATTER.parse(value));
```

Comment:
no sense converting it between String -> LocalDateTime -> String -> ZonedDateTime -> LocalDateTime -> String when we can just feed an `Instant`